clients/privacy-center/cypress/e2e/consent-events.cy.ts

```diff
@@ -0,0 +1,571 @@
+    });
+  }
+
+  describe("when banner_and_modal experience", () => {
+    beforeEach(() => {
+      // Load the banner_and_modal experience
+      const fixture = "experience_banner_modal.json";
+      cy.fixture(`consent/${fixture}`).then((data) => {
+        let experience = data.items[0];
+        stubConfig({ experience });
+      });
+    });
+
+    it("should fire FidesEvents for all key interactions", () => {
+      const expectedEvents: FidesEventExpectation[] = [];
+
+      // Initialize and show banner on page load
+      expectedEvents.push(
+        { type: "FidesInitializing", detail: {} },
+        { type: "FidesInitialized", detail: {} },
+        {
+          type: "FidesUIShown",
+          detail: { extraDetails: { servingComponent: "banner" } },
+        },
+      );
+
+      // Open modal from banner button
+      cy.get("#fides-banner .fides-manage-preferences-button").click();
+      expectedEvents.push({
+        type: "FidesUIShown",
+        detail: { extraDetails: { servingComponent: "modal" } },
+      });
+
+      // Toggle first notice on then off
+      cy.get("#fides-modal .fides-toggle-input").first().click().click();
+      expectedEvents.push(
+        {
+          type: "FidesUIChanged",
+          detail: {
+            extraDetails: {
+              servingComponent: "modal",
+              trigger: {
+                type: "toggle",
+                label: "Advertising",
+                id: "advertising",
+                checked: true,
+              },
+            },
+          },
+        },
+        {
+          type: "FidesUIChanged",
+          detail: {
+            extraDetails: {
+              servingComponent: "modal",
+              trigger: {
+                type: "toggle",
+                label: "Advertising",
+                id: "advertising",
+                checked: false,
+              },
+            },
+          },
+        },
+      );
+
+      // Toggle second notice off (Analytics opt out)
+      cy.get("#fides-modal .fides-toggle-input").eq(1).click();
+      expectedEvents.push({
+        type: "FidesUIChanged",
+        detail: {
+          extraDetails: {
+            servingComponent: "modal",
+            trigger: {
+              type: "toggle",
+              label: "Analytics",
+              id: "analytics_opt_out",
+              checked: false,
+            },
+          },
+        },
+      });
+
+      // Save current preference selections
+      cy.get("#fides-modal .fides-save-button").click();
+      expectedEvents.push(
+        { type: "FidesModalClosed", detail: {} },
+        {
+          type: "FidesUpdating",
+          detail: {
+            extraDetails: {
+              // servingComponent: "modal",
+              consentMethod: "save",
+              // trigger: {
+              //   type: "button",
+              //   label: "Save",
+              //   id: "save-preferences",
+              // },
```

Comment:
can we clean up these commented lines? seeing this in several places in this test. Are we planning to bring these back in at some point?

Reply:
So, I'd like to have the events include these properties. However, today our Updated/Updating events are completely disjointed from the buttons that trigger them, so it was really gnarly to try to include these.

In other words the tests only pass if these are commented out, but that's because our events are currently pretty inconsistent.

I don't think I'll have time to fix this tech debt quite yet, I tried for a bit and got nervous about the changes!

Reply:
Maybe just a follow up ticket with a comment here then?

Reply:
Done!